source/slang/slang-ir-legalize-image-subscript.cpp

```diff
@@ -0,0 +1,196 @@
+#include "slang-ir-legalize-image-subscript.h"
+
+#include "slang-ir.h"
+#include "slang-ir-insts.h"
+#include "slang-ir-util.h"
+#include "slang-ir-clone.h"
+#include "slang-ir-specialize-address-space.h"
+#include "slang-parameter-binding.h"
+#include "slang-ir-legalize-varying-params.h"
+
+namespace Slang
+{
+    void legalizeImageSubscriptStore(TargetRequest* target, IRBuilder& builder, IRInst* storeInst, DiagnosticSink* sink)
+    {
+        builder.setInsertBefore(storeInst);
+        auto getElementPtr = as<IRGetElementPtr>(storeInst->getOperand(0));
+        IRImageSubscript* imageSubscript = nullptr;
```

Comment:
```
        auto getElementPtr = as<IRGetElementPtr>(storeInst->getOperand(0));
        IRImageSubscript* imageSubscript = nullptr;
        if (getElementPtr)
            imageSubscript = as<IRImageSubscript>(getElementPtr->getBase());
        else
            imageSubscript = as<IRImageSubscript>(storeInst->getOperand(0));
```

if can be simplified as
```
IRImageSubscript* imageSubscript = nullptr;
if (auto getElementPtr = as<IRGetElementPtr>(storeInst->getOperand(0)))
{}
else
{}
```

Reply:
`getElementPtr` is used in a later scope, I cannot move it into an inline if assignment

Reply:
I see.